src/st-flash/flash.c

```diff
@@ -51,13 +60,13 @@ int main(int ac, char** av) {
         return(-1);
     }
 
-    printf("st-flash %s\n", STLINK_VERSION);
+    ILOG("st-flash %s\n", STLINK_VERSION);
 
     sl = stlink_open_usb(o.log_level, 1, (char *)o.serial, o.freq);
 
     if (sl == NULL) { return(-1); }
 
-    if (sl->flash_type == STLINK_FLASH_TYPE_UNKNOWN) {
+    if (STLINK_FLASH_TYPE_UNKNOWN == sl->flash_type) {
```

Comment:
Same question regarding PR scope, why that change here ? these are purely debatable style change

Reply:
Yeah, well true. AFAIK the previous state was the most common type in this toolset.

Reply:
I can remove these changes, they are not related to this PR, it's just something I'm used to do to prevent compiler warnings

Reply:
What sort of compiler warnings? It may be something reasonable to consider.

Reply:
Below there is a link to the Travis CI build:
https://travis-ci.org/github/stlink-org/stlink/builds/705314574?utm_source=github_status&utm_medium=notification

It is giving the compiler warnings I was talking about:
[ 55%] Building C object CMakeFiles/st-flash.dir/src/st-flash/flash_opts.c.o
/home/travis/build/stlink-org/stlink/src/st-flash/flash_opts.c:298:29: error: 
      equality comparison with extraneous parentheses
      [-Werror,-Wparentheses-equality]
        } else if ((o->area == FLASH_OTP)) {
                    ~~~~~~~~^~~~~~~~~~~~
/home/travis/build/stlink-org/stlink/src/st-flash/flash_opts.c:298:29: note: 
      remove extraneous parentheses around the comparison to silence this
      warning
        } else if ((o->area == FLASH_OTP)) {
                   ~        ^           ~
/home/travis/build/stlink-org/stlink/src/st-flash/flash_opts.c:298:29: note: 
      use '=' to turn this equality comparison into an assignment
        } else if ((o->area == FLASH_OTP)) {
                            ^~
                            =

If by accident you would write:
FLASH_OTP = o->area (iso FLASH_OTP == o->area), the compiler would be able to catch that since you obviously can't assign a value to a #define. 
As it is now, o->area = FLASH_OTP is valid so the compiler can't help when you accidentally remove a '='.
It takes some getting used to in reading the comparison, but it helps.

Reply:
I need to look at this again to gain some understanding why this actually happens.

Reply:
Maybe you should remove the inner brackets ( ):
`else if (o->area == FLASH_OTP)`
as this seems to compare (o->area == FLASH_OTP) == TRUE
Can you try this out?

Reply:
Thanks, I missed that one, that solved the build warnings